h2o-core/src/main/java/water/persist/PersistHex.java

```diff
@@ -0,0 +1,100 @@
+package water.persist;
+
+import water.DKV;
+import water.H2O;
+import water.Key;
+import water.Value;
+import water.fvec.C1NChunk;
+import water.util.FrameUtils;
+import water.util.Log;
+
+import java.io.*;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PersistHex extends Persist {
+
+  static final String HEX_PATH_PREFIX = PersistManager.Schemes.HEX + "://";
+
+  Key<?> fromHexPath(String path) {
+    Key<?> key = Key.make(path.substring(HEX_PATH_PREFIX.length()));
+    if (! key.isChunkKey()) {
+      throw new IllegalArgumentException("Only Chunk keys are supported for HEX schema");
+    }
+    return key;
+  }
+
+  InputStream open(Key<?> key) {
+    byte[] bytes = ((C1NChunk) DKV.getGet(key)).getBytes();
+    return new ByteArrayInputStream(bytes);
+  }
+
+  @Override
+  public InputStream open(String path) {
+    return open(fromHexPath(path));
+  }
+
+  @Override
+  public OutputStream create(String path, boolean overwrite) {
+    Key<?> ck = fromHexPath(path);
+    return new ByteChunkOutputStream(ck);
+  }
+
+  private static class ByteChunkOutputStream extends ByteArrayOutputStream {
+    private final Key<?> _chunkKey;
+
+    public ByteChunkOutputStream(Key<?> chunkKey) {
+      super();
+      _chunkKey = chunkKey;
+    }
+
+    @Override
+    public void close() throws IOException {
+      super.close();
```

Comment:
Question: Closing might throw IOException. Better put it into DKV in a finally block, provided it is a priority. Should we attempt to put the resulting bytes in the DKV, even if the underlying stream is not closed ?

```java
      try {
        super.close();
      } finally {
        byte[] myBytes = toByteArray();
        DKV.put(_chunkKey, new Value(_chunkKey, new C1NChunk(myBytes)));
      }
```

Reply:
The underlying stream is ByteArrayOutputStream and it will never through an exception, it is a NOOP.

If it did through an exception, I think it is correct to just propagate it and don't change the state of DKV.